lib/ansible/plugins/action/gather_facts.py

```diff
@@ -107,10 +116,21 @@ def run(self, tmp=None, task_vars=None):
 
             self._remove_tmp_path(self._connection._shell.tmpdir)
         else:
-            # do it async
+            # do it async, aka parallel
             jobs = {}
+
             for fact_module in modules:
                 mod_args = self._get_module_args(fact_module, task_vars)
+
+                #  if module does not handle timeout, use timeout to handle module, hijack async_val as this is what async_wrapper uses
+                # TODO: make this action compain about async/async settings, use parallel option instead .. or remove parallel in favor of async settings?
+                if timeout and 'gather_timeout' not in mod_args:
+                    self._task.async_val = int(timeout)
+                elif async_val != 0:
+                    self._task.async_val = async_val
+                else:
+                    self._task.async_val = 0
```

Comment:
this was just me being paranoid in case we do eventually enable 'async gather_facts' .. which is not hard when 1 module is involved (lots of fun when 2+ jids are involved!).